sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusConnection.cs

```diff
@@ -0,0 +1,533 @@
+        ///   connection string.  For example, ";EntityPath=telemetry-hub".
+        ///
+        ///   If you have defined a shared access policy directly on the Service Bus entity itself, then copying the connection string from that
+        ///   Service Bus entity will result in a connection string that contains the name.
+        /// </remarks>
+        ///
+        public ServiceBusConnection(string connectionString) : this(connectionString, null, connectionOptions: null)
+        {
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="ServiceBusConnection"/> class.
+        /// </summary>
+        ///
+        /// <param name="connectionString">The connection string to use for connecting to the Service Bus namespace; it is expected that the entity name and the shared key properties are contained in this connection string.</param>
+        /// <param name="connectionOptions">A set of options to apply when configuring the connection.</param>
+        ///
+        /// <remarks>
+        ///   If the connection string is copied from the Service Bus namespace, it will likely not contain the name of the desired Service Bus entity,
+        ///   which is needed.  In this case, the name can be added manually by adding ";EntityPath=[[ Service Bus entity NAME ]]" to the end of the
+        ///   connection string.  For example, ";EntityPath=telemetry-hub".
+        ///
+        ///   If you have defined a shared access policy directly on the Service Bus entity itself, then copying the connection string from that
+        ///   Service Bus entity will result in a connection string that contains the name.
+        /// </remarks>
+        ///
+        public ServiceBusConnection(string connectionString,
+                                  ServiceBusConnectionOptions connectionOptions) : this(connectionString, null, connectionOptions)
+        {
+        }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="ServiceBusConnection"/> class.
+        /// </summary>
+        ///
+        /// <param name="connectionString">The connection string to use for connecting to the Service Bus namespace; it is expected that the shared key properties are contained in this connection string, but not the Service Bus entity name.</param>
+        /// <param name="entityName">The name of the specific entity to associate the connection with.</param>
+        ///
+        /// <remarks>
+        ///   If the connection string is copied from the Service Bus entity itself, it will contain the name of the desired Service Bus entity,
+        ///   and can be used directly without passing the <paramref name="entityName" />.  The name of the Service Bus entity should be
+        ///   passed only once, either as part of the connection string or separately.
+        /// </remarks>
+        ///
+        public ServiceBusConnection(string connectionString, string entityName)
```

Comment:
nit:  Formatting is inconsistent for the constructors.    Since you seem to largely be keeping the Event Hubs classes, they should have one parameter per line, with the first parameter being on the same line as the method/constructor and the rest being indented to that level.

Reply:
This is different than what we do in Storage, where we have each parameter be on its own line including the first parameter and to indent from the start of method signature.

Do you think we should adopt this guideline for Service Bus? I would think we would want to be consistent across libraries.

Reply:
I don't think it matters, personally, so long as it is consistent within a service.   At the end of the day, I don't think there's a loss of readability if each service has a slightly different format, which is what the goal would be.    The contributors to Storage aren't likely to be jumping over to Service Bus in the same workstream nor vice versa.